js/source/esri-tile-pyramid.js

```diff
@@ -0,0 +1,321 @@
+'use strict';
+
+var Tile = require('mapbox-gl/js/source/tile');
+var TileCoord = require('mapbox-gl/js/source/tile_coord');
+var Point = require('mapbox-gl/node_modules/point-geometry');
+var Cache = require('mapbox-gl/js/util/mru_cache');
+var util = require('mapbox-gl/js/util/util');
+
+module.exports = TilePyramid;
+
+function TilePyramid(options) {
```

Comment:
This and the rest of the new functionality needs JSDoc documentation information to match the rest of mapbox-gl-js.